File: app/src/main/java/ru/binaryblitz/sportup/custom/CustomMapFragment.java
package ru.binaryblitz.sportup.custom;

import android.content.Context;
import android.os.Bundle;
import android.os.SystemClock;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import com.google.android.gms.maps.SupportMapFragment;

import ru.binaryblitz.sportup.activities.EventsMapActivity;

public class CustomMapFragment {
    public static class TouchableWrapper extends FrameLayout {

        private long lastTouched = 0;
        private static final long SCROLL_TIME = 200L;
        private UpdateMapAfterUserInteraction updateMapAfterUserInteraction;

        public TouchableWrapper(Context context) {
            super(context);
            try {
                updateMapAfterUserInteraction = (EventsMapActivity) context;
            } catch (ClassCastException e) {
                throw new ClassCastException();
            }
        }

        @Override
        public boolean dispatchTouchEvent(MotionEvent ev) {
            switch (ev.getAction()) {
                case MotionEvent.ACTION_DOWN:
                    lastTouched = SystemClock.uptimeMillis();
                    break;
                case MotionEvent.ACTION_UP:
                    final long now = SystemClock.uptimeMillis();
                    if (now - lastTouched > SCROLL_TIME) {
                        updateMapAfterUserInteraction.onUpdateMapAfterUserInteraction();
                    }
                    break;
            }
            return super.dispatchTouchEvent(ev);
        }

        public interface UpdateMapAfterUserInteraction {
            void onUpdateMapAfterUserInteraction();
        }
    }

    public static class MySupportMapFragment extends SupportMapFragment {
        public View mOriginalContentView;
        public TouchableWrapper mTouchView;

        @Override
        public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState) {
            mOriginalContentView = super.onCreateView(inflater, parent, savedInstanceState);
            mTouchView = new TouchableWrapper(getActivity());
            mTouchView.addView(mOriginalContentView);
            return mTouchView;
        }

        @Override
        public View getView() {
            return mOriginalContentView;
        }
    }
}

File: app/src/main/java/ru/binaryblitz/sportup/custom/CustomMapFragment.kt
package ru.binaryblitz.sportup.custom

import android.content.Context
import android.os.Bundle
import android.os.SystemClock
import android.view.LayoutInflater
import android.view.MotionEvent
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout

import com.google.android.gms.maps.SupportMapFragment

import ru.binaryblitz.sportup.activities.EventsMapActivity

class CustomMapFragment {
    class TouchableWrapper(context: Context) : FrameLayout(context) {

        private var timeLastTouched: Long = 0
        private var updateMapAfterUserInteraction: UpdateMapAfterUserInteraction? = null

        init {
            try {
                updateMapAfterUserInteraction = context as EventsMapActivity
            } catch (e: ClassCastException) {
                throw ClassCastException()
            }

        }

        override fun dispatchTouchEvent(motionEvent: MotionEvent): Boolean {
            when (motionEvent.action) {
                MotionEvent.ACTION_DOWN -> timeLastTouched = SystemClock.uptimeMillis()
                MotionEvent.ACTION_UP -> {
                    val now = SystemClock.uptimeMillis()
                    if (now - timeLastTouched > SCROLL_TIME) {
                        updateMapAfterUserInteraction!!.onUpdateMapAfterUserInteraction()
                    }
                }
            }
            return super.dispatchTouchEvent(motionEvent)
        }

        interface UpdateMapAfterUserInteraction {
            fun onUpdateMapAfterUserInteraction()
        }

        companion object {
            private val SCROLL_TIME = 200L
        }
    }

    class MySupportMapFragment : SupportMapFragment() {
        lateinit var originalContentView: View
        lateinit var touchView: TouchableWrapper

        override fun onCreateView(inflater: LayoutInflater?, parent: ViewGroup?, savedInstanceState: Bundle?): View? {
            originalContentView = super.onCreateView(inflater, parent, savedInstanceState)!!
            touchView = TouchableWrapper(activity)
            touchView.addView(originalContentView)
            return touchView
        }

        override fun getView(): View? {
            return originalContentView
        }
    }
}
